storm-core/src/jvm/org/apache/storm/metric/ClusterMetricsConsumerExecutor.java

```diff
@@ -40,31 +39,49 @@ public ClusterMetricsConsumerExecutor(String consumerClassName, Object registrat
     public void prepare() {
         try {
             metricsConsumer = (IClusterMetricsConsumer)Class.forName(consumerClassName).newInstance();
+            metricsConsumer.prepare(registrationArgument);
         } catch (Exception e) {
-            throw new RuntimeException("Could not instantiate a class listed in config under section " +
-                    Config.STORM_CLUSTER_METRICS_CONSUMER_REGISTER + " with fully qualified name " + consumerClassName, e);
-        }
+            LOG.error("Could not instantiate or prepare Cluster Metrics Consumer with fully qualified name " +
+                    consumerClassName, e);
 
-        metricsConsumer.prepare(registrationArgument);
+            if (metricsConsumer != null) {
+                metricsConsumer.cleanup();
+            }
+            metricsConsumer = null;
+        }
     }
 
     public void handleDataPoints(final IClusterMetricsConsumer.ClusterInfo clusterInfo, final Collection<DataPoint> dataPoints) {
-        try {
-            metricsConsumer.handleDataPoints(clusterInfo, dataPoints);
-        } catch (Throwable e) {
-            LOG.error("Error while handling cluster data points, consumer class: " + consumerClassName, e);
+        if (metricsConsumer == null) {
```

Comment:
do we need to attempt to re-initialize the metricsConsumer in handleDataPoints. Given that this method gets executed in a schedule-recurring loop. 
If the metrics reporter isn't initialized it means that either the user is mis-configured or didn't supply the required class. Any of these errors means user need to fix and restart the nimbus. 
So in that case it would be save to initialize the metrics reporter only once on nimbus start and we can probably add a debug message in the handleDataPoints if metricsConsumer is null so that users can figure out if the metrics are not getting published.